[RFR] Put take

ThieryMichel opened the pull request:
follow #9

ThieryMichel committed:
add put effect

ThieryMichel committed:
add take effect

ThieryMichel committed:
update README

ThieryMichel committed:
add spawn effect

ThieryMichel committed:
add fork effect

ThieryMichel committed:
add parentEventEmitter to properly propagate event

djhi reviewed (changes requested):
Can you split this in at least two pr:
- one for refactoring
- another for the new effects

It's currently hard to review and we shouldn't mix refactoring and new features

ThieryMichel committed:
update readme

djhi commented on README.md:
````diff
@@ -76,10 +76,33 @@ Calls a continuation passing style function.
+```
+
+### spawn
+Launch another sg generator, but do not wait for it to end. Returning a function that will resolve when the generator is done instead.
````
s/end. Returning/end, returning/

sedy-bot committed:
Typo fix s/end. Returning/end, returning/

As requested by @djhi at https://github.com/marmelab/sg/pull/8#discussion_r92127453

djhi commented on src/effects/fork.js:
```diff
@@ -0,0 +1,11 @@
+import createEffect from './createEffect';
+
+export const handleForkEffect = sgImpl => ([callable, ...args], parentEmitter, emitter) => {
+    const promise = sgImpl(callable, emitter)(...args);
```
It doesn't check for callable to be a generator as done in [spawn](https://github.com/marmelab/sg/pull/8/files#diff-0e50efdcb840bb7943ea324e136720f6R6) ?

ThieryMichel replied:
I'ts spawn that should not verify, since sg does it.

ThieryMichel committed:
remove generator verification from spawn effect, since sg already does it
